### PR TITLE
[MAD][DATA1018][lab-intro-to-ml] Maris Font

### PR DESCRIPTION
**Pull request by marisfont**
Hey Belen - Finally finished the lab. 

I want to thank you for your support and understanding yesterday and today. Starting Machine Learning with the flu and a fever wasn't ideal, but I wouldn't have made it through without you!!!

THANK YOUUUUUUU

### CODE REVIEW

File: module-3/lab-intro-to-ml/your-code/main.ipynb [hunk 1]
```diff
@@ -2522,7 +2522,7 @@
   },
```
Comment (Gobinde43) [inline module-3/lab-intro-to-ml/your-code/main.ipynb]: - "Hint: you may use a loop to change one column at a time but it is more efficient to use apply." Using apply can be something like the following:
```Python
austin[wrong_type_columns] = austin[wrong_type_columns].apply(lambda col: pd.to_numeric(col, errors='coerce'))
```
- "Next, count the number of rows of austin and missing_values."  Use another method called as well as len in the following and check the differences.
```Python
#Yours
austin_rows = len(austin)
print(austin_rows)
missing_values_rows = len(missing_values)
print(missing_values_rows)
#size method
print(austin.size)
print(missing_values.size)
```
- "As you can see from the output, the majority of missing data is in one column called PrecipitationSumInches. What's the number of missing values in this column in ratio to its total number of rows?"  Try something like the following:
```Python
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- "In the cell below, convert the Date column values from string to numeric values using toordinal()." You can do something as simple as the following:
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- "Your y should be a subset of austin_fixed containing one column TempAvgF." Try to not write down one by one the column labels. You can do something like the following:
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```
- Try to train also SVM rbf with gamma and C parameters and check the differences in the performance metrics. Find below two links to a quite clear explanation of SVM (rbf and poly) gamma and c parameters. 
The default in SVC is rbf. 
https://www.youtube.com/watch?v=m2a2K4lprQw
https://www.youtube.com/watch?v=joTa_FeMZ2s
- When training classifiers it's a good idea to use some other performance metrics. For instance,confusion matrix and precision score among others. Find below some links that could be interesting. 
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html
https://towardsdatascience.com/beyond-accuracy-precision-and-recall-3da06bea9f6c
